modules/core/src/main/scala/notebook/NBSerializer.scala

```diff
@@ -218,44 +234,45 @@ object NBSerializer extends Logging {
 
   implicit val worksheetFormat = Json.format[Worksheet]
 
-  case class Notebook(
-    metadata: Option[Metadata] = None,
-    cells: Option[List[Cell]] = Some(Nil),
-    worksheets: Option[List[Worksheet]] = None,
-    autosaved: Option[List[Worksheet]] = None,
-    nbformat: Option[Int]) {
-    def name = metadata.map(_.name).getOrElse("Anonymous")
-  }
-
   implicit val notebookFormat = Json.format[Notebook]
 
   def fromJson(json: JsValue): Option[Notebook] = {
     logTrace("\r\n**************************************\r\n")
     logTrace(Json.prettyPrint(json))
     logTrace("**************************************\r\n")
     json.validate[Notebook] match {
-      case s: JsSuccess[Notebook] => {
+      case s: JsSuccess[Notebook] =>
         s.get match {
-          case Notebook(None,None,None,None,None) =>
+          case Notebook(None, None, None, None, None) =>
             logWarn("Nothing in the notebook data.")
-            None
-          case notebook =>
-            Some( notebook.cells.map { _ => notebook } getOrElse notebook.copy(cells = Some(Nil)) )
+            throw new NotebookDeserializationError("Got an empty notebook")
```

Comment:
maybe a good idea - at  least we wouldn't have two ways at the same time `Option[Notebook] + exception`.
but this would be quite a API breaking change...

P.S. an exception is `deal`able too - sometimes more fancy scala features proves dangerous, but this one is likely OK ;)